compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -280,12 +280,17 @@ static void checkRegisterSpilling(IREE::HAL::ExecutableVariantOp &variantOp,
 
   if (!llvm::offloading::amdgpu::getAMDGPUMetaDataFromImage(
           llvm::MemoryBufferRef(obj, ""), infoMap, abiVersion)) {
-    for (const auto &[_, metaData] : infoMap) {
+    for (llvm::StringMapEntry<llvm::offloading::amdgpu::AMDGPUKernelMetaData>
+             &entry : infoMap) {
+      StringRef dispatchName = entry.getKey();
+      llvm::offloading::amdgpu::AMDGPUKernelMetaData &metaData =
+          entry.getValue();
```

Comment:
This is quite verbose, I think the previous impl with structured bindings was more readable...